presto-main/src/main/java/com/facebook/presto/execution/scheduler/FixedSubsetNodeSetSupplier.java

```diff
@@ -0,0 +1,340 @@
+                    allNodesByHost.put(host, node);
+                }
+                catch (UnknownHostException e) {
+                    // ignore
+                }
+            }
+
+            return new NodeSet(
+                    activeNodesByNodeId.build(),
+                    activeWorkersByNetworkPath.build(),
+                    nodeManager.getCoordinators(),
+                    activeNodes,
+                    allNodes,
+                    allNodesByHost.build(),
+                    allNodesByHostAndPort.build(),
+                    consistentHashingNodeProvider);
+        };
+    }
+
+    @Override
+    public void close()
+            throws IOException
+    {
+        scheduledFuture.cancel(true);
+    }
+
+    public PriorityQueue<NodeSetAcquireRequest> getPendingRequests()
+    {
+        return pendingRequests;
+    }
+
+    public Map<QueryId, List<InternalNode>> getQueryNodesMap()
+    {
+        return queryNodesMap;
+    }
+
+    public Map<QueryId, Supplier<QueryState>> getQueryStateMap()
+    {
+        return queryStateMap;
+    }
+
+    public static class NodeSetAcquireRequest
+            implements Comparable
+    {
+        private QueryId queryId;
+        private int count;
+
+        private Supplier<QueryState> queryStateSupplier;
+        public CompletableFuture<?> getCompletableFuture()
+        {
+            return completableFuture;
+        }
+
+        public void setCompletableFuture(CompletableFuture<?> completableFuture)
+        {
+            this.completableFuture = completableFuture;
+        }
+
+        private CompletableFuture<?> completableFuture;
+        public NodeSetAcquireRequest(
+                QueryId queryId,
+                int count,
+                Supplier<QueryState> queryStateSupplier)
+        {
+            this.queryId = queryId;
+            this.count = count;
+            this.queryStateSupplier = queryStateSupplier;
+        }
+
+        public Supplier<QueryState> getQueryStateSupplier()
+        {
+            return queryStateSupplier;
+        }
+
+        public void setQueryStateSupplier(Supplier<QueryState> queryStateSupplier)
+        {
+            this.queryStateSupplier = queryStateSupplier;
+        }
+
+        public QueryId getQueryId()
+        {
+            return queryId;
+        }
+
+        public void setQueryId(QueryId queryId)
+        {
+            this.queryId = queryId;
+        }
+
+        public int getCount()
+        {
+            return count;
+        }
+
+        public void setCount(int count)
+        {
+            this.count = count;
+        }
+
+        @Override
+        public int compareTo(@NotNull Object other)
+        {
+            NodeSetAcquireRequest otherRequest = (NodeSetAcquireRequest) other;
+            if (otherRequest.getCount() == this.getCount()) {
+                return 0;
+            }
+            return this.getCount() < otherRequest.getCount() ? -1 : 1;
```

Comment:
QueryIds contain timestamps (or maybe we should explicitly include create timestamps here) -- to break the ties when the hash-partition count is the same (which will probably happen all of the time).

Reply:
Yes. I presumed we don't need that for POC